### ISSUE TITLE
Rewrite debug visulisation with vulkan

### ISSUE THREAD

**Issue by NotStirred** [OpenCubicChunks/CubicChunks#531]

**Comment by AidanLovelace** [comment]
Make sure that the window is created from the main thread if you want this to work on MacOS. The OGL vis crashes the entire process (you can't even get the java stacktrace) on MacOS because it's creating the window on a different thread. (I believe it's the Server thread)

**Comment by NotStirred** [comment]
Yes that was a known issue with the opengl implementation, we just don't have a mac user around to actually test everything

**Comment by AidanLovelace** [comment]
Here's one!

**Comment by NotStirred** [comment]
@Barteks2x any chance you could have a look at getting your opengl implementation working on mac?

**Comment by AidanLovelace** [comment]
After doing some investigation, it appears it is running on the render thread. The window must be *created* on the main thread, but it can be rendered from any thread.

**Comment by Niko-sk2x** [comment]
Shouldn't be too hard. The entire reason it didn't work in the first implementation was that lamost everything was off thread but that didn't work on windows either

**Comment by AidanLovelace** [comment]
I'm working on it right now and making progress
